modules/date/strptime-tz.c

```diff
@@ -513,7 +513,14 @@ again:		switch (c = *fmt++) {
                                                  NULL, 2);
                                 if (ep != NULL) {
 					tm->tm_isdst = i;
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#include <sys/param.h>
+#if defined(BSD)
+					*tm_gmtoff = -(tm->tm_gmtoff);
+#else
 					*tm_gmtoff = -(timezone);
+#endif
+#endif
 					*tm_zone = tzname[i];
```

Comment:
This should be extracted into a new function.

long _get_gmtoff_from_struct_tm(const struct tm *tm);